chihacknight/decarbonize-my-state
# Generate All 3.5K Power Plant Social Images

## Overview

Adds new social card images for **all** power plants, which look like this:

![image](https://user-images.githubusercontent.com/3187531/235932291-417e9a78-903c-49a5-8970-53eca278fffb.png)

This process involved a few steps:

1. **Creating a new power plant social card route that renders the card** - this was based off of the SocialCardPage and takes a `state` and `plant` param, like `/power-plant-social-card/?state=new_york&plant=ravenswood-generating-station`
2. **Update the generate-social-cards script to also generate power plants** - this actually was the hardest part of the work, since power plants number about 3.5 thousand, so I added enhancements to the script like command line arguments that let you opt into a debugging mode with more logging and browser previews (`--debugging`), and a headless flag (`--headless) that you can combo with debugging. Also updated the `README` and `package.json` with new helpers and explanations
3. **Ran the script against Netlify** - I made this draft PR, got a Netlify preview, updated the `SiteDomain` in `generate-social-cards.js` to that domain, and then ran `yarn generate-social-power-plants`. This was needed because `gatsby develop` always eventually bombs out due to memory issues  😖 
4. **Copy the output directory into static**

And that's it! 🎉 

Closes #200

### Demo

| Console Command Output with `--debugging` flag | Social Card URL |
| --- | --- |
| ![Screenshot from 2023-05-03 08-32-54](https://user-images.githubusercontent.com/3187531/235931263-80d43a36-bdf4-4799-a7c2-da1f22086e7e.png) ![Screenshot from 2023-05-03 08-23-24](https://user-images.githubusercontent.com/3187531/235931304-21a009cc-6518-48bc-9533-12d848e36be3.png) | ![Screenshot from 2023-05-03 08-33-59](https://user-images.githubusercontent.com/3187531/235931889-f763b5c7-d7c6-4541-b8c5-c7a3b9d16c15.png) ![Screenshot from 2023-05-03 08-35-05](https://user-images.githubusercontent.com/3187531/235931843-de7eb176-7bd5-4aba-8e86-cf2098f57fb9.png) |

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

1. Test the actual social generation scripts and making sure you can run them and understand them - if not, I need to tweak the documentation
2. Confirm a power plant page has a `<meta name="image"` tag with a `content` pointing to a valid exported social card image (but using the prod `decarbmystate.com` domain rather than `localhost`).
